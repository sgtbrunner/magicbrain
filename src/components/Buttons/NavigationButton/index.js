import NavigationButton from './navigation-button.component';

export default NavigationButton;
