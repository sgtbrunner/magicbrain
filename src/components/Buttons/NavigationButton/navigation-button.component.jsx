import React from 'react';
import PropTypes from 'prop-types';

const NavigationButton = ({ onClick, children }) => (
  <button className="f4 link black pv2 ma3 b--none bg-transparent" onClick={onClick} type="button">
    {children}
  </button>
);

NavigationButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
};

export default NavigationButton;
