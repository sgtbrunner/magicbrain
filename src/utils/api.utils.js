const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
