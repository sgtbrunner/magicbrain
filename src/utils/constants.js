export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
