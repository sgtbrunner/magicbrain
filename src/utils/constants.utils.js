const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
