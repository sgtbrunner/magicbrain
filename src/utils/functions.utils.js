export const getUserFromLocalStorage = () => localStorage.getItem('user');
