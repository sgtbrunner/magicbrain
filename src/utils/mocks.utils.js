export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
