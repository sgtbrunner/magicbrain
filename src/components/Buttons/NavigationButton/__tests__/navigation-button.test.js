import { screen } from '@testing-library/react';

import { mountComponent } from '../../../../utils/test.utils';
import NavigationButton from '../navigation-button.component';

const HOME = 'Home';
const onClick = jest.fn();
const props = { children: HOME, onClick };

const mountNavigationButtonComponent = () => mountComponent({ component: NavigationButton, props });

describe('NavigationButton component', () => {
  it('should match snapshot', () => {
    const { container } = mountNavigationButtonComponent();

    expect(container).toMatchSnapshot();
  });

  it(`should render 'Home' as button text if children prop is 'Home'`, () => {
    mountNavigationButtonComponent();

    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent(HOME);
  });
});
