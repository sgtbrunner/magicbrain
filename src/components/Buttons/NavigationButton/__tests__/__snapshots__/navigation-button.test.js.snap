// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButton component should match snapshot 1`] = `
<div>
  <button
    class="f4 link black pv2 ma3 b--none bg-transparent"
    type="button"
  >
    Home
  </button>
</div>
`;
